### PR TITLE
Use mint and withdraw + slither cleanup

### PR DESCRIPTION
**Pull request by hieronx**

### CODE REVIEW

File: src/interfaces/token/ITranche.sol [hunk 1]
```diff
@@ -13,7 +13,6 @@ interface IERC1404 {
 interface ITranche is IERC20Metadata, IERC7575Share, IERC1404 {
     // --- Events ---
     event File(bytes32 indexed what, address data);
-    event File(bytes32 indexed what, address data1, address data2);
```
Comment (hieronx) [inline src/interfaces/token/ITranche.sol:16]: Was not used

File: src/libraries/TransientStorage.sol [hunk 2]
```diff
@@ -15,13 +15,13 @@ library TransientStorage {
         }
     }
 
-    function tloadAddress(bytes32 slot) internal returns (address value) {
+    function tloadAddress(bytes32 slot) internal view returns (address value) {
```
Comment (hieronx) [inline src/libraries/TransientStorage.sol:18]: To fix compiler warning

File: src/token/Tranche.sol [hunk 3]
```diff
@@ -105,13 +110,12 @@ contract Tranche is ERC20, ITranche {
     }
 
     function _onTransfer(address from, address to, uint256 value) internal {
-        if (hook != address(0)) {
-            require(
-                IHook(hook).onERC20Transfer(from, to, value, HookData(hookDataOf(from), hookDataOf(to)))
+        require(
```
Comment (hieronx) [inline src/token/Tranche.sol:113]: A bit cleaner

File: test/integration/Redeem.t.sol [hunk 4]
```diff
@@ -261,7 +261,6 @@ contract RedeemTest is BaseTest {
         ERC7540Vault vault = ERC7540Vault(vault_);
         ITranche tranche = ITranche(address(vault.share()));
         deposit(vault_, investor, amount, false); // request and execute deposit, but don't claim
-        uint256 investorBalanceBefore = erc20.balanceOf(investor);
```
Comment (hieronx) [inline test/integration/Redeem.t.sol:264]: Was not used

File: src/libraries/BitmapLib.sol [hunk 5]
```diff
@@ -31,11 +31,6 @@ library BitmapLib {
         return uint128(value) << shift;
     }
 
-    /// @notice Concatenate uint64 values to create a uint128 value
```
Comment (hieronx) [inline src/libraries/BitmapLib.sol:34]: Was not used

**Review by AStox — DISMISSED** [review]

**Review by ilinzweilin — APPROVED** [review]

**Pull request merged**